### PR TITLE
Add a platform support document for Android

### PR DESCRIPTION
**Pull request by chriswailes**
r? @joshtriplett

### CODE REVIEW

**Review by jyn514 — COMMENTED** [review]
Thanks for writing this up!

File: src/doc/rustc/src/platform-support/android.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+[Android] is a mobile operating system built on top of the LInux kernel.
+
+[Android]: https://source.android.com/
+
+## Target maintainers
+
+- Chris Wailes ([@chriswailes](https://github.com/chriswailes))
+- Matthew Maurer ([@maurer](https://github.com/maurer))
+- Martin Geisler
```
[the remark this answers is not in the document]
Comment (chriswailes) [reply src/doc/rustc/src/platform-support/android.md]: GitHub link added.  Thanks for catching that.

**Comment by sanxiyn** [comment]
You also need to edit `SUMMARY.md` (for mdBook) and `platform-support.md` (for links). See #92098 for an example.

File: src/doc/rustc/src/platform-support/android.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# *-linux-android and *-linux-androideabi
+
+**Tier: 2**
+
+[Android] is a mobile operating system built on top of the LInux kernel.
```
Comment (fbstj) [inline src/doc/rustc/src/platform-support/android.md:5]:
```suggestion
[Android] is a mobile operating system built on top of the Linux kernel.
```

Comment (chriswailes) [reply src/doc/rustc/src/platform-support/android.md:5]: Done.  Thanks for catching that.

**Comment by chriswailes** [comment]
> You also need to edit `SUMMARY.md` (for mdBook) and `platform-support.md` (for links). See #92098 for an example.

Thanks for pointing that out.

**Comment by chriswailes** [comment]
The NDK update policy language in this document is based on discussions held on [Zulip](https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/Formalizing.20Rust's.20Android.20NDK.2FAPI.20Support/near/291907407) and in-person at RustConf with Josh Triplett.

File: src/doc/rustc/src/SUMMARY.md [hunk 3]
```diff
@@ -23,6 +23,7 @@
     - [armv6k-nintendo-3ds](platform-support/armv6k-nintendo-3ds.md)
     - [armv7-unknown-linux-uclibceabi](platform-support/armv7-unknown-linux-uclibceabi.md)
     - [armv7-unknown-linux-uclibceabihf](platform-support/armv7-unknown-linux-uclibceabihf.md)
+    - [\*-android(eabi)?](platform-support/android.md)
```
Comment (joshtriplett) [inline src/doc/rustc/src/SUMMARY.md:26]: This is a really minor nit, but while this document uses `*` (which is a fairly widely understood wildcard), it doesn't use any form of regular expression syntax. Can I suggest just using literally `*-android and *-androideabi`?

```suggestion
    - [\*-android and \*android-eabi](platform-support/android.md)
```

File: src/doc/rustc/src/platform-support/android.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- Martin Geisler ([@mgeisler](https://github.com/mgeisler))
+
+## Requirements
+
+This target is cross-compiled from a host environment. Development may be done
+from the [source tree] or using the Android NDK.
+
+[source tree]: https://source.android.com/docs/setup/build/downloading
+
+Android targets support std and generated binaries use the ELF file format.
```
Comment (joshtriplett) [inline src/doc/rustc/src/platform-support/android.md:22]:
```suggestion
Android targets support std. Generated binaries use the ELF file format.
```

File: src/doc/rustc/src/platform-support/android.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+## NDK/API Update Policy
+
+Rust will support the most recent Long Term Support (LTS) Android Native
+Development Kit (NDK).  By default Rust will support all API levels supported
+by the NDK but a higher minimum API level may be required if deemed necessary.
```
Comment (joshtriplett) [inline src/doc/rustc/src/platform-support/android.md:28]:
```suggestion
by the NDK, but a higher minimum API level may be required if deemed necessary.
```

File: src/doc/rustc/src/SUMMARY.md (hunk 3 above)
Comment (chriswailes) [reply src/doc/rustc/src/SUMMARY.md:26]: Done

File: src/doc/rustc/src/platform-support/android.md (hunk 4 above)
Comment (chriswailes) [reply src/doc/rustc/src/platform-support/android.md:22]: Done

File: src/doc/rustc/src/platform-support/android.md (hunk 5 above)
Comment (chriswailes) [reply src/doc/rustc/src/platform-support/android.md:28]: Done

**Pull request merged**